internal/librarian/release_notes.go

```diff
@@ -237,6 +237,46 @@ func findLatestGenerationCommit(repo gitrepo.Repository, state *config.Librarian
 	return res, nil
 }
 
+// groupByIDAndSubject aggregates conventional commits for ones have the same Piper ID and subject in the footer.
+func groupByIDAndSubject(commits []*conventionalcommits.ConventionalCommit) []*conventionalcommits.ConventionalCommit {
+	idToCommits := make(map[string][]*conventionalcommits.ConventionalCommit)
+	var singletons []*conventionalcommits.ConventionalCommit
+	for i := range len(commits) {
+		commit := commits[i]
+		// a commit is not considering for grouping if it doesn't have a footer or
+		// the footer doesn't have a Piper ID.
+		if commit.Footers == nil {
+			commit.Footers = make(map[string]string)
+			commit.Footers["Library-IDs"] = commit.LibraryID
+			singletons = append(singletons, commit)
+			continue
+		}
+
+		id, ok := commit.Footers["PiperOrigin-RevId"]
+		if !ok {
+			commit.Footers["Library-IDs"] = commit.LibraryID
+			singletons = append(singletons, commit)
```

Comment:
These can just be directly added to the response instead of creating the singletons slice.

Reply:
Done.